File: implementations/java/src/SQLiteBatch.java

import java.io.File;
import java.sql.*;

public class SQLiteBatch {

    public static void main(String[] args) {
        String dbPath = String.format("jdbc:sqlite:%s/database.db",
                new File(SQLiteBatch.class.getProtectionDomain().getCodeSource().getLocation().getPath())
                        .getParent());
        int i, hash;
        try (
            Connection conn = DriverManager.getConnection(dbPath);
            Statement stmt = conn.createStatement();
            PreparedStatement ps = conn.prepareStatement(
                "INSERT INTO tb_java_batch(hash) VALUES (?)"
            );
        ) {
            stmt.executeUpdate("DROP TABLE IF EXISTS tb_java_batch;"
                    + "CREATE TABLE tb_java_batch"
                    + "(id INTEGER PRIMARY KEY AUTOINCREMENT,"
                    + " hash INTEGER NOT NULL)");

            conn.setAutoCommit(false);

            for (i = 1; i <= 10000; i++) {
                hash = Hash.hash32(i);
                ps.setInt(1, hash);
                ps.addBatch();
            }

            ps.executeBatch();

            conn.commit();

        } catch (Exception e) {
            System.err.println(e.getMessage());
        }
    }
}

File: implementations/py/src/sqlite-batch.py
from hashes import hash32
from contextlib import closing
from pathlib import Path
from sqlite3 import connect

dbPath = Path(__file__).parent.parent / "database.db"

try:
    with closing(connect(dbPath)) as conn:
        with conn:
            conn.executescript(
                """DROP TABLE IF EXISTS tb_py_batch;
                CREATE TABLE tb_py_batch
                (id INTEGER PRIMARY KEY AUTOINCREMENT,
                hash INTEGER NOT NULL);"""
            )
            conn.commit()

            buf = []
            for i in range(1, 10_001):
                hash = hash32(i)
                buf.append(f"INSERT INTO tb_py_batch(hash) VALUES ({hash});")

            query = "BEGIN;\n" + "\n".join(buf) + "\nCOMMIT;"
            conn.executescript(query)

except Exception as e:
    print(e)
